### PR TITLE
feat: remove invariant check and apply cap to rebate factor

### PR DESCRIPTION
**Pull request by cdummett**
... based off slack [thread](https://vegaprotocol.slack.com/archives/CAHA5EX0F/p1723631490236589) and core PR https://github.com/vegaprotocol/vega/pull/11578.

PR removes invariance check and instead caps the rebate factor to a maximum which is the sum of the treasury and buy back fee factors.

Maximum should be recalculated and applied on governance proposals updating factors.

### CODE REVIEW

File: protocol/0095-HVMR-high_volume_maker_rebate.md [hunk 1]
```diff
@@ -108,8 +112,20 @@ As the rebate possible level interacts with other fee settings there must be a r
...
+
+1. If the `buyback_fee_factor` is reduced through a proposal in the middle of an epoch such that a parties `additional_rebate_factor` is less than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to their current `additional_rebate_factor`. (<a name="0095-HVMR-033" href="#0095-HVMR-033">0095-HVMR-033</a>).
+1. If the `treasury_fee_factor` is reduced through a proposal in the middle of an epoch such that a parties `additional_rebate_factor` is greater than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to their current `additional_rebate_factor`. (<a name="0095-HVMR-034" href="#0095-HVMR-034">0095-HVMR-034</a>).
+1. If the `buyback_fee_factor` and `treasury_fee_factor` are both reduced through a batch proposal in the middle of an epoch such that a parties current `additional_rebate_factor` is greater than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to their current `additional_rebate_factor`. (<a name="0095-HVMR-035" href="#0095-HVMR-035">0095-HVMR-035</a>).
```
Comment (Jiajia-Cui) [inline protocol/0095-HVMR-high_volume_maker_rebate.md]: is `effective_additional_rebate_factor` not meant to be reset here?

File: protocol/0095-HVMR-high_volume_maker_rebate.md [hunk 2]
```diff
@@ -108,8 +112,20 @@ As the rebate possible level interacts with other fee settings there must be a r
...
+1. If the `buyback_fee_factor` and `treasury_fee_factor` are both increased through a batch proposal in the middle of an epoch such that a parties `additional_rebate_factor` is greater than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to the new maximum (`buyback_fee_factor` + `treasury_fee_factor`). (<a name="0095-HVMR-032" href="#0095-HVMR-032">0095-HVMR-032</a>).
+
+1. If the `buyback_fee_factor` is reduced through a proposal in the middle of an epoch such that a parties `additional_rebate_factor` is less than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to their current `additional_rebate_factor`. (<a name="0095-HVMR-033" href="#0095-HVMR-033">0095-HVMR-033</a>).
+1. If the `treasury_fee_factor` is reduced through a proposal in the middle of an epoch such that a parties `additional_rebate_factor` is greater than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to their current `additional_rebate_factor`. (<a name="0095-HVMR-034" href="#0095-HVMR-034">0095-HVMR-034</a>).
```
Comment (Jiajia-Cui) [inline protocol/0095-HVMR-high_volume_maker_rebate.md]:
```suggestion
1. If the `treasury_fee_factor` is reduced through a proposal in the middle of an epoch such that a party's `additional_rebate_factor` is greater than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to their current `additional_rebate_factor`. (<a name="0095-HVMR-034" href="#0095-HVMR-034">0095-HVMR-034</a>).
```

Comment (Jiajia-Cui) [reply protocol/0095-HVMR-high_volume_maker_rebate.md]: in fact all the "parties" should be "party's"

File: protocol/0095-HVMR-high_volume_maker_rebate.md (hunk 1 above)
Comment (cdummett) [reply protocol/0095-HVMR-high_volume_maker_rebate.md]: Ah thanks there is a typo in 034 and 035. Should say if rebate is less than the factors - fixing now.

File: protocol/0095-HVMR-high_volume_maker_rebate.md [hunk 3]
```diff
@@ -108,8 +112,20 @@ As the rebate possible level interacts with other fee settings there must be a r
...
 1. If a party does not qualify for the lowest tier, their `additional_rebate_factor`is set to `0` (<a name="0095-HVMR-028" href="#0095-HVMR-028">0095-HVMR-028</a>).
+
+#### Capping the effective rebate factor
+
+1. If a party qualifies for a tier where the `additional_rebate_factor` is greater than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be capped to a maximum and set to (`buyback_fee_factor` + `treasury_fee_factor`). In this case no treasury or buyback fees should be collected by the network for trades involving this party. (<a name="0095-HVMR-029" href="#0095-HVMR-029">0095-HVMR-029</a>).
+
+1. If the `buyback_fee_factor` is increased through a proposal in the middle of an epoch such that a parties `additional_rebate_factor` is greater than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to the new maximum (`buyback_fee_factor` + `treasury_fee_factor`). (<a name="0095-HVMR-030" href="#0095-HVMR-030">0095-HVMR-030</a>).
```
Comment (Jiajia-Cui) [inline protocol/0095-HVMR-high_volume_maker_rebate.md]:
```suggestion
1. If the `buyback_fee_factor` is increased through a proposal in the middle of an epoch such that a party's `additional_rebate_factor` is greater than (`buyback_fee_factor` + `treasury_fee_factor`), their `effective_additional_rebate_factor` should be set to the new maximum (`buyback_fee_factor` + `treasury_fee_factor`). (<a name="0095-HVMR-030" href="#0095-HVMR-030">0095-HVMR-030</a>).
```

File: protocol/0095-HVMR-high_volume_maker_rebate.md [hunk 4]
```diff
@@ -104,12 +108,24 @@ As the rebate possible level interacts with other fee settings there must be a r
...
-1. Given two parties making markets in two separate spot markets using quote assets with different quantum values, if the parties generated equal volume (e.g. 10,000 USD) then they should both have a maker volume fraction of `0.5`. (<a name="0095-HVMR-024" href="#0095-HVMR-024">0095-HVMR-024</a>).
-1. Given two parties making markets in a separate derivative and spot market using a settlement and quote asset with different quantum values respectively, if the parties generated equal volume (e.g. 10,000 USD) then they should both have a maker volume fraction of `0.5`. (<a name="0095-HVMR-025" href="#0095-HVMR-025">0095-HVMR-025</a>).
+1. Given two party's making markets in two separate derivative markets using settlement assets with different quantum values, if the party's generated equal volume (e.g. 10,000 USD) then they should both have a maker volume fraction of `0.5`. (<a name="0095-HVMR-023" href="#0095-HVMR-023">0095-HVMR-023</a>).
```
Comment (Jiajia-Cui) [inline protocol/0095-HVMR-high_volume_maker_rebate.md]:
```suggestion
1. Given two parties' making markets in two separate derivative markets using settlement assets with different quantum values, if the party's generated equal volume (e.g. 10,000 USD) then they should both have a maker volume fraction of `0.5`. (<a name="0095-HVMR-023" href="#0095-HVMR-023">0095-HVMR-023</a>).
```

File: protocol/0095-HVMR-high_volume_maker_rebate.md [hunk 5]
```diff
@@ -104,12 +108,24 @@ As the rebate possible level interacts with other fee settings there must be a r
...
+1. Given two party's making markets in two separate spot markets using quote assets with different quantum values, if the party's generated equal volume (e.g. 10,000 USD) then they should both have a maker volume fraction of `0.5`. (<a name="0095-HVMR-024" href="#0095-HVMR-024">0095-HVMR-024</a>).
```
Comment (Jiajia-Cui) [inline protocol/0095-HVMR-high_volume_maker_rebate.md]:
```suggestion
1. Given two parties' making markets in two separate spot markets using quote assets with different quantum values, if the party's generated equal volume (e.g. 10,000 USD) then they should both have a maker volume fraction of `0.5`. (<a name="0095-HVMR-024" href="#0095-HVMR-024">0095-HVMR-024</a>).
```

File: protocol/0095-HVMR-high_volume_maker_rebate.md [hunk 6]
```diff
@@ -104,12 +108,24 @@ As the rebate possible level interacts with other fee settings there must be a r
...
+1. Given two party's making markets in a separate derivative and spot market using a settlement and quote asset with different quantum values respectively, if the party's generated equal volume (e.g. 10,000 USD) then they should both have a maker volume fraction of `0.5`. (<a name="0095-HVMR-025" href="#0095-HVMR-025">0095-HVMR-025</a>).
```
Comment (Jiajia-Cui) [inline protocol/0095-HVMR-high_volume_maker_rebate.md]:
```suggestion
1. Given two parties' making markets in a separate derivative and spot market using a settlement and quote asset with different quantum values respectively, if the party's generated equal volume (e.g. 10,000 USD) then they should both have a maker volume fraction of `0.5`. (<a name="0095-HVMR-025" href="#0095-HVMR-025">0095-HVMR-025</a>).
```

**Review by Jiajia-Cui — DISMISSED** [review]

**Review by Jiajia-Cui — DISMISSED** [review]

**Review by Jiajia-Cui — DISMISSED** [review]

**Review by Jiajia-Cui — DISMISSED** [review]

**Review by davidsiska-vega — APPROVED** [review]

**Pull request merged**